### ISSUE TITLE
NUnit throws an exception when run under UWP built with the .NET native toolchain

### ISSUE THREAD

**Issue by LaPeste** [nunit/nunit#3970]
I am working on running NUnit tests under UWP. It all works fine when the application is built with the previous .NET toolchain. However, the same is not true when the application is built with the .NET native toolchain, where the following exception is raised:

```
// exception
System.PlatformNotSupportedException: 'GetRuntimeInterfaceMap() is not supported on this runtime.'

// stacktrace
   at System.Reflection.Runtime.TypeInfos.RuntimeTypeInfo.GetInterfaceMap(Type interfaceType) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.Reflection.Core\src\System\Reflection\Runtime\TypeInfos\RuntimeTypeInfo.cs:line 198
   at NUnit.Framework.Constraints.Comparers.EquatablesComparer.GetEquatableImplementations(Type type)
   at NUnit.Framework.Constraints.Comparers.EquatablesComparer.FirstImplementsIEquatableOfSecond(Type first, Type second)
   at NUnit.Framework.Constraints.Comparers.EquatablesComparer.Equal(Object x, Object y, Tolerance& tolerance, ComparisonState state)
   at NUnit.Framework.Constraints.NUnitEqualityComparer.AreEqual(Object x, Object y, Tolerance& tolerance, ComparisonState state)
   at NUnit.Framework.Constraints.NUnitEqualityComparer.AreEqual(Object x, Object y, Tolerance& tolerance, Boolean topLevelComparison)
   at NUnit.Framework.Constraints.NUnitEqualityComparer.AreEqual(Object x, Object y, Tolerance& tolerance)
   at NUnit.Framework.Constraints.EqualConstraint.ApplyTo[TActual](TActual actual)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression, String message, Object[] args)
   at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression)
   at NUnit.Framework.Tests.TestWithComparableClass.EqualityTest() in C:\Users\andre\Development\NUnit_UWPnative_repro\NUnit_UWPnative_repro.UWP\TestWithComparableClass.cs:line 23
   at _$ILCT$.$ILT$ReflectionDynamicInvoke$.InvokeRetV(Object thisPtr, IntPtr methodToCall, ArgSetupState& argSetupState, Boolean targetIsThisCall)
   at System.InvokeUtils.CalliIntrinsics.Call(IntPtr dynamicInvokeHelperMethod, Object thisPtr, IntPtr methodToCall, ArgSetupState& argSetupState, Boolean isTargetThisCall)
   at System.InvokeUtils.CallDynamicInvokeMethod(Object thisPtr, IntPtr methodToCall, Object thisPtrDynamicInvokeMethod, IntPtr dynamicInvokeHelperMethod, IntPtr dynamicInvokeHelperGenericDictionary, Object targetMethodOrDelegate, Object[] parameters, BinderBundle binderBundle, Boolean wrapInTargetInvocationException, Boolean invokeMethodHelperIsThisCall, Boolean methodToCallIsThisCall)
```

The exception is raised by [this line](https://github.com/nunit/nunit/blob/d8e06e82210733f887b9afb9056ef18b7258b9cf/src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs#L64).
Basically, it'd be great if a workaround for the lack of `GetRuntimeInterfaceMap()` in UWP native could be found.
</br>
</br>
As a side note, to run NUnit tests in UWP I use [nunit.xamarin](https://www.nuget.org/packages/nunit.xamarin/). This also raises an exception at line 27 of `MainPage.xaml.cs` of the attach repro project. Don't worry about that exception as it's only coming from the lib trying to automatically add tests. In order to do so, it also uses an unsupported method from the reflection API. The tests are manually added at line 29 of `MainPage.xaml.cs`, so it's fine not to worry about this nunit.xamarin exception. Just keep going until you reach the `EqualityTest`.
</br>
</br>

[NUnit_UWPnative_repro.zip](https://github.com/nunit/nunit/files/7396967/NUnit_UWPnative_repro.zip)
